x/programs/rust/sdk-macros/src/lib.rs

```diff
@@ -170,14 +184,61 @@ pub fn public(_: TokenStream, item: TokenStream) -> TokenStream {
         }
     };
 
+    let inputs: Punctuated<FnArg, Token![,]> = binding_args_props.collect();
+    let args = inputs.iter().skip(1).map(|arg| match arg {
+        FnArg::Typed(PatType { pat, .. }) => pat,
+        _ => unreachable!(),
+    });
+    let name = name.to_string();
+
+    let return_type = match &input.sig.output {
+        ReturnType::Type(_, ty) => ty.as_ref().clone(),
+        ReturnType::Default => parse_quote!(()),
+    };
+
+    let block = Box::new(parse_quote! {{
+        param_0
+            .program()
+            .call_function::<#return_type, _>(#name, (#(#args),*), param_0.max_units())
+            .expect("calling the external program failed")
```

Comment:
This is only here temporarily... we need to force returning results for `public` functions.